src/walletSdk/Utils/soroban.ts

```diff
@@ -165,3 +165,83 @@ export const parseTokenAmount = (
 
   return BigInt(parsed.toString());
 };
+
+// This function attempt to convert soroban contract values
+// to common types like string, array, buffer, JSON string, etc.
+export const scValByType = (scVal: xdr.ScVal) => {
```

Comment:
We probably should add module for Soroban in the future, considering that the codebase should grow for it 🤔